pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -624,6 +698,42 @@ func (m *kubeGenericRuntimeManager) killContainersWithSyncResult(pod *v1.Pod, ru
 	return
 }
 
+func (m *kubeGenericRuntimeManager) preStopSidecars(pod *v1.Pod, containers []*kubecontainer.Container) {
+	wg := sync.WaitGroup{}
+	wg.Add(len(containers))
+
+	for _, container := range containers {
+		go func(container *kubecontainer.Container) {
+			defer wg.Done()
+			gracePeriod := int64(minimumGracePeriodInSeconds)
```

Comment:
We should do this check after the `if pod != nil` condition below; because only after that condition do we know that `pod` is not nil. In this current implementation, we may hit a nil pointer access if pod happens to be nil.

Reply:
fixed